chains/solana/contracts/programs/ccip-router/src/token_context.rs

```diff
@@ -13,6 +13,35 @@ pub struct TokenAdminRegistry {
     pub administrator: Pubkey,
     pub pending_administrator: Pubkey,
     pub lookup_table: Pubkey,
+    // binary representation of indexes that are writable in token pool lookup table
+    // lookup table can store 256 addresses
+    pub writable_indexes: [u128; 2],
+}
+
+impl TokenAdminRegistry {
+    // set writable inserts bits from left to right
+    // index 0 is left-most bit
+    pub fn set_writable(&mut self, index: u8) {
+        match index < 128 {
+            true => {
+                self.writable_indexes[0] |= 1 << (127 - index);
+            }
+            false => {
+                self.writable_indexes[1] |= 1 << (255 - index);
+            }
+        }
+    }
+
+    pub fn is_writable(&self, index: u8) -> bool {
+        match index < 128 {
+            true => self.writable_indexes[0] & 1 << (127 - index) != 0,
+            false => self.writable_indexes[1] & 1 << (255 - index) != 0,
+        }
+    }
+
+    pub fn reset_writable(&mut self) {
+        self.writable_indexes = [0, 0];
+    }
 }
```

Comment:
Up to you whether to leave this here for now or not, but in general I'd rather have functions defined in the versioned logic modules instead of here.